pkg/backup/compaction_test.go

```diff
@@ -1005,39 +939,152 @@ func getNumBackupsInRestore(t *testing.T, db *sqlutils.SQLRunner, jobID jobspb.J
 
 // ensureBackupExists ensures that a backup exists that spans the given start and end times.
 func ensureBackupExists(
-	t *testing.T, db *sqlutils.SQLRunner, collectionURIs string, start, end int64, opts string,
+	t *testing.T,
+	db *sqlutils.SQLRunner,
+	collectionURI []string,
+	start, end hlc.Timestamp,
+	opts *tree.ShowBackupOptions,
 ) {
 	t.Helper()
-	showBackupQ := fmt.Sprintf(`SHOW BACKUP FROM LATEST IN (%s)`, collectionURIs)
-	if opts != "" {
-		showBackupQ += " WITH " + opts
-	}
+	startTime, err := strconv.ParseFloat(start.AsOfSystemTime(), 64)
+	require.NoError(t, err)
+	endTime, err := strconv.ParseFloat(end.AsOfSystemTime(), 64)
+	require.NoError(t, err)
+	startNano, endNano := int64(startTime), int64(endTime)
+
 	// Convert times to millisecond epoch. We compare millisecond epoch instead of
 	// nanosecond epoch because the backup time is stored in milliseconds, but timeutil.Now()
 	// will return a nanosecond-precise epoch.
 	times := db.Query(t,
 		fmt.Sprintf(`SELECT DISTINCT
 		COALESCE(start_time::DECIMAL * 1e6, 0),
 		COALESCE(end_time::DECIMAL * 1e6, 0)
-		FROM [%s]`, showBackupQ),
-	)
+		FROM [%s]`, showBackupQuery(collectionURI, opts),
+		))
 	defer times.Close()
 	found := false
 	for times.Next() {
 		var startTime, endTime int64
 		require.NoError(t, times.Scan(&startTime, &endTime))
-		if startTime == start/1e3 && endTime == end/1e3 {
+		if startTime == startNano/1e3 && endTime == endNano/1e3 {
 			found = true
 			break
 		}
 	}
-	require.True(t, found, "missing backup with start time %d and end time %d", start, end)
+	require.True(t, found, "missing backup with start time %d and end time %d", startNano, endNano)
 }
 
-// getTime returns the current time in nanoseconds since epoch.
-func getTime(t *testing.T) int64 {
+func getTime() hlc.Timestamp {
+	return hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}
+}
+
+// Some named variables to improve test readability.
+const fullCluster = ""
+
+var noAOST = hlc.Timestamp{}
+
+// fullBackupQuery creates a `BACKUP INTO` statement.
+func fullBackupQuery(
+	t *testing.T,
+	targets string,
+	collectionURI []string,
+	aost hlc.Timestamp,
+	opts *tree.BackupOptions,
```

Comment:
uber nit: if you wanted to make this even easier on the test writer and reader, opts could be a string: e.g. this is valid: "WITH new_db_name=foo", but this is not "new_db_name=foo". I.e. the caller needs to specify the with clause. That way, a user doesn't need to init a tree every time they use options. You could then define `noOpts=""`

Reply:
Ah the reason I went with this route is that in `validateCompactedBackupForTables`, we can compose the `ShowBackupOptions` from the provided options, which isn't so easy with a string.

Reply:
Tried it out, yeah I think I agree with you. Updated!